sources/networkmanager/vpc-attachment.go

```diff
@@ -0,0 +1,88 @@
+package networkmanager
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/networkmanager"
+	"github.com/aws/aws-sdk-go-v2/service/networkmanager/types"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+func vpcAttachmentGetFunc(ctx context.Context, client *networkmanager.Client, scope, query string) (*types.VpcAttachment, error) {
+	out, err := client.GetVpcAttachment(ctx, &networkmanager.GetVpcAttachmentInput{
+		AttachmentId: &query,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return out.VpcAttachment, nil
+}
+
+func vpcAttachmentItemMapper(scope string, awsItem *types.VpcAttachment) (*sdp.Item, error) {
+	attributes, err := sources.ToAttributesCase(awsItem)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// The uniqueAttributeValue for this is a nested value of AttachmentId:
+	if awsItem != nil && awsItem.Attachment != nil {
+		attributes.Set("attachmentId", *awsItem.Attachment.AttachmentId)
+	}
+
+	item := sdp.Item{
+		Type:            "networkmanager-vpc-attachment",
+		UniqueAttribute: "attachmentId",
+		Attributes:      attributes,
+		Scope:           scope,
+		LinkedItemQueries: []*sdp.LinkedItemQuery{
+			{
+				Query: &sdp.Query{
+					// +overmind:link networkmanager-core-network
+					// Search for all vpc attachments with this core network
+					Type:   "networkmanager-core-network",
+					Method: sdp.QueryMethod_GET,
+					Query:  *awsItem.Attachment.CoreNetworkId,
```

Comment:
Will this always exist? If it's unset we'll get a nil pointer exception, might be better to wrap in an if statement